Symfony/CS/Tests/Fixer/Symfony/SpacesAfterSemicolonFixerTest.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Tests\Fixer\Symfony;
+
+use Symfony\CS\Tests\Fixer\AbstractFixerTestBase;
+
+/**
+ * @author SpacePossum
+ */
+class SpacesAfterSemicolonFixerTest extends AbstractFixerTestBase
+{
+    /**
+     * @dataProvider provideCases
+     */
+    public function testFix($expected, $input = null)
+    {
+        $this->makeTest($expected, $input);
+    }
+
+    public function provideCases()
+    {
+        return array(
+            array(
+                '<?php
+                    test();
+                    $a; // test
+                ',
+            ),
+            array(
+                '<?php test();     ',
+            ),
+            array(
+                '<?php
+                    test();     // test
+                ',
+            ),
+            array(
+                '<?php test();       /* */ //',
+            ),
+            array(
+                '<?php
+                    test(); $a = 4;
+                ',
+                '<?php
+                    test();     $a = 4;
+                ',
+            ),
+            array(
+                '<?php
+                    test(); $a = 4;
+                ',
+                '<?php
+                    test();$a = 4;
+                ',
+            ),
+            array(
+        '<?php
+        // 0 0 0
+        for (; ; ) {
```

Comment:
TBH I get lost here. Maybe it would be more readable if we provide a separate test for each `for` notation?